examples/chat/src/network.rs

```diff
@@ -3,154 +3,114 @@ use std::time::Duration;
 
 use libp2p::futures::prelude::*;
 use libp2p::swarm::{NetworkBehaviour, Swarm, SwarmEvent};
-use libp2p::{dcutr, gossipsub, identify, identity, noise, ping, relay, yamux, PeerId};
+use libp2p::{
+    dcutr, gossipsub, identify, identity, mdns, noise, ping, relay, rendezvous, yamux, PeerId,
+};
 use multiaddr::Multiaddr;
 use tokio::sync::{mpsc, oneshot};
 use tokio::time;
-use tracing::{debug, error, info, trace, warn};
+use tracing::{debug, error, trace, warn};
 
 pub mod client;
 pub mod events;
 pub mod types;
 
 use client::NetworkClient;
 
-const PROTOCOL_VERSION: &str = concat!("/", env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
+const PROTOCOL_VERSION: &str = concat!("/", "boot-node", "/", "0.5.0");
 
 #[derive(NetworkBehaviour)]
 struct Behaviour {
     dcutr: dcutr::Behaviour,
     identify: identify::Behaviour,
     gossipsub: gossipsub::Behaviour,
+    mdns: mdns::tokio::Behaviour,
     ping: ping::Behaviour,
-    relay_client: relay::client::Behaviour,
+    rendezvous: rendezvous::client::Behaviour,
+    relay: relay::client::Behaviour,
 }
 
 pub async fn run(
     keypair: identity::Keypair,
     port: u16,
-    relay_address: Multiaddr,
+    rendezvous_namespace: rendezvous::Namespace,
+    relay_addresses: Vec<Multiaddr>,
+    rendezvous_addresses: Vec<Multiaddr>,
 ) -> eyre::Result<(NetworkClient, mpsc::Receiver<types::NetworkEvent>)> {
-    let (client, mut event_receiver, event_loop) = init(keypair).await?;
+    let mut rendezvous = HashMap::new();
+    for address in rendezvous_addresses.clone().into_iter() {
+        let entry = match peak_peer_id(&address) {
+            Ok(peer_id) => (peer_id, RendezvousEntry::new(address)),
```

Comment:
clone individual items instead of the whole `Vec`, especially since some of the addrs won't be used / will cause an error

```suggestion
    for address in &rendezvous_addresses {
        let entry = match peak_peer_id(&address) {
            Ok(peer_id) => (peer_id, RendezvousEntry::new(address.clone())),
```